File: Python/15.py
class Solution(object):
    def threeSum(self, nums):
        lst = []
        nums.sort()
        for i in range(len(nums)):
            left,right = i+1,len(nums)-1
            while(left < right):
                total = nums[i] + nums[left] + nums[right]
                if total == 0 and [nums[i],nums[left],nums[right]] not in lst:
                    lst.append([nums[i],nums[left],nums[right]])
                    left += 1
                    right -= 1
                else:
                    if total >= 0:
                        right -= 1
                    else:
                        left += 1
        return lst

File: Java/15.java
import java.util.*;
class Solution {
    public List<List<Integer>> threeSum(int[] nums) {
        Arrays.sort(nums);
        List<List<Integer>> lst = new ArrayList<>();
        List<Integer> arr = new ArrayList<>();
        int left, right, total;
        for(int i = 0; i < nums.length; i++){
            if(i>0 && nums[i] == nums[i-1]) continue;
            left = i+1;
            right = nums.length-1;
            while(left < right){
                total = nums[i] + nums[left] + nums[right];
                if(total == 0){
                    arr = new ArrayList<>(Arrays.asList(nums[i], nums[left], nums[right]));
                    lst.add(arr);
                    left++;
                    right--;
                    while(left < right && nums[left] == nums[left-1]) left++;
                    while(left < right && nums[right] == nums[right+1]) right--;
                }
                else if(total < 0) left++;
                else right--;
            }
        }
        return lst;
    }
}